extension/src/lib/portBinding.js

```diff
@@ -13,51 +13,334 @@
+                args: ["{that}", "{arguments}.0"]
+            },
+            "onIncomingRead.handle": {
+                listener: "{that}.handleMessage",
+                args: [gpii.chrome.portBinding.type.READ_RECEIPT, "{that}.handleRead", "{arguments}.0"]
+            },
+            "onIncomingReadReceipt.handle": "{that}.handleReceipt",
+            "onIncomingWrite.handle": {
+                listener: "{that}.handleMessage",
+                args: [gpii.chrome.portBinding.type.WRITE_RECEIPT, "{that}.handleWrite", "{arguments}.0"]
+            },
+            "onIncomingWriteReceipt.handle": "{that}.handleReceipt"
         },
         invokers: {
-            postMessage: {
-                funcName: "gpii.chrome.portBinding.postMessage",
+            read: {
+                funcName: "gpii.chrome.portBinding.postRequest",
+                args: ["{that}", gpii.chrome.portBinding.type.READ, "{arguments}.0"]
+            },
+            write: {
+                funcName: "gpii.chrome.portBinding.postRequest",
+                args: ["{that}", gpii.chrome.portBinding.type.WRITE, "{arguments}.0"]
+            },
+            postReceipt: {
+                funcName: "gpii.chrome.portBinding.postReceipt",
+                args: ["{that}", "{arguments}.0", "{arguments}.1", "{arguments}.2", "{arguments}.3"]
+            },
+            setPort: {
+                funcName: "gpii.chrome.portBinding.setPort",
+                args: ["{that}.options.connectionName", "{that}.id"]
+            },
+            handleMessage: {
+                funcName: "gpii.chrome.portBinding.handleMessage",
+                args: ["{that}", "{arguments}.0", "{arguments}.1", "{arguments}.2"]
+            },
+            handleReceipt: {
+                funcName: "gpii.chrome.portBinding.handleReceipt",
                 args: ["{that}", "{arguments}.0"]
-            }
+            },
+            // handleRead and handleWrite must be implemented by an integrator. They will be called by the handleMessage
+            // flow for handling read and write requests. They may return a promise to indicate if the task has
+            // completed successfully or not. If a promise isn't returned all responses will be treated as a success.
+            handleRead: "fluid.notImplemented",
+            handleWrite: "fluid.notImplemented"
         }
     });
 
+    /**
+     * An object which allows two way communication with other pages.
+     * See: https://developer.chrome.com/apps/runtime#type-Port
+     *
+     * @typedef {Object} Port
+     */
+
+    /**
+     * Creates a connection and returns a port.
+     *
+     * @param {String} connectionName - the name of the connection. Will be used as the prefix for the name passed
+     *                                  to the connection.
+     * @param {String} id - the id of the connection. Will be used as the suffix for the name passed to the conneciton.
```

Comment:
Typo conneciton. I don't find the documentation for these two members very clear, since they use the word "connection" in two different senses. Perhaps the overall name "connectionName" isn't well chosen? I still think you could do more to explain what this name means. I think if you are referring to simply the "Port" object from the runtime API you should continue to use that term to avoid confusion, and then perhaps free up the name "connection" to refer to an abstraction which you layer on top of it.